### PR TITLE
fix: 404 error page responsive and test case

### PR DESCRIPTION
**Pull request by abhishekjakhar**
## Hey, I just made a Pull Request!

- Improve the responsive version of 404 Error page
- There were no assertions in test cases
- Unnecessary scroll bar was coming on desktop and mobile, that has been fixed
- Mic drop image was having alt attribute but there was no value given to it, that has been given

##### Before

error-page-before

##### After

error-page-after

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/core/src/layout/ErrorPage/ErrorPage.tsx [hunk 1]
```diff
@@ -48,9 +55,9 @@ export const ErrorPage = ({
   const navigate = useNavigate();
 
   return (
-    <Grid container className={classes.container}>
+    <Grid className={classes.container}>
```
Comment (freben) [inline packages/core/src/layout/ErrorPage/ErrorPage.tsx:58]: I believe that you are expected to have either the container or item prop on each Grid element, right?

File: packages/core/src/layout/ErrorPage/MicDrop.tsx [hunk 2]
```diff
@@ -18,16 +18,29 @@ import React from 'react';
...
 import MicDropSvgUrl from './mic-drop.svg';
 
-const useStyles = makeStyles({
+const useStyles = makeStyles(theme => ({
   micDrop: {
     maxWidth: '60%',
-    bottom: 10,
-    right: 10,
+    bottom: 16,
+    right: 16,
```
Comment (freben) [inline packages/core/src/layout/ErrorPage/MicDrop.tsx:25]: Ooh this is an opportunity to make these `theme.spacing(2)` instead for consistency

Comment (abhishekjakhar) [reply packages/core/src/layout/ErrorPage/MicDrop.tsx]: Done

File: packages/core/src/layout/ErrorPage/ErrorPage.tsx (hunk 1 above)
Comment (abhishekjakhar) [reply packages/core/src/layout/ErrorPage/ErrorPage.tsx]: Giving `<Grid/>` a container prop was making unwanted scroll on both web and mobile. Now, I have added **container** prop but with it I have also added **spacing** prop with the value 0. It is working fine now even with container prop.

**Review by stefanalund — APPROVED** [review]

**Pull request merged**